validation-test/BuildSystem/extractsymbols-no-file-to-process.test

```diff
@@ -0,0 +1,18 @@
+# REQUIRES: standalone_build
+# REQUIRES: OS=macosx
+
+# Ensure the logic for extracting symbols does not fail is there is nothing to process
```

Comment:
Can you add a better description of what this is actually testing? From our conversation it sounds like what this is really about is that the symbol extraction actually runs code even when we perform a dry-run and that previously this caused the dry-run code to fail (IIRC grep was specifically mentioned as the failing command). Can you expand the comment and add that context for when we both forget = p?

I would also change the name of this to extractsymbols-dry-run-does-not-fail.test or whatever works for you. Lets just go more for verbosity b/c people are not going to look at this code for a while.

Reply:
Both points are addressed in b903a6cc658511ea333b357f05cbfd6362f17a17